aspnetcore/mvc/controllers/actions.md

```diff
@@ -14,80 +14,91 @@ uid: mvc/controllers/actions
 
-**Redirect**
+#### 2. Methods resulting in a non-empty response body with a predefined content type
 
-Returns a redirect to another action or destination (using `Redirect`, `LocalRedirect`, `RedirectToAction` or `RedirectToRoute`). Example: `return RedirectToAction("Complete", new {id = 123});`
+Most helper methods in this category include a `ContentType` property, allowing you to set the `Content-Type` response header to describe the response body.
 
-In addition to the methods above, an action can also simply return an object. In this case, the object will be formatted based on the client's request. Learn more about [Formatting Response Data](../models/formatting.md)
+There are two result types within this category: [View](xref:mvc/views/overview) and [Formatted Response](xref:mvc/models/formatting).
 
-### Cross-Cutting Concerns
+* **View**
+
+    This type returns a view which uses a model to render HTML. For example, `return View(customer);` passes a model to the view for data-binding.
+
+* **Formatted Response**
 
-In most apps, many actions will share parts of their workflow. For instance, most of an app might be available only to authenticated users, or might benefit from caching. When you want to perform some logic before or after an action method runs, you can use a *filter*. You can help keep your actions from growing too large by using [Filters](filters.md) to handle these cross-cutting concerns. This can help eliminate duplication within your actions, allowing them to follow the [Don't Repeat Yourself (DRY) principle](http://deviq.com/don-t-repeat-yourself/).
+    This type returns JSON or a similar data exchange format to represent an object in a specific manner. For example, `return Json(customer);` serializes the provided object into JSON format.
+    
+    Other common methods of this type include `File`, `PhysicalFile`, and `VirtualFile`. For example, `return PhysicalFile(customerFilePath, "text/xml");` returns an XML file described by a `Content-Type` response header value of "text/xml".
+
+#### 3. Methods resulting in a non-empty response body formatted in a content type negotiated with the client
+
+This category is better known as **Content Negotiation**. [Content negotiation](xref:mvc/models/formatting#content-negotiation) applies whenever an action returns an [ObjectResult](https://docs.microsoft.com/aspnet/core/api/microsoft.aspnetcore.mvc.objectresult) type or something other than an [IActionResult](https://docs.microsoft.com/aspnet/core/api/microsoft.aspnetcore.mvc.iactionresult) implementation. An action that returns a non-`IActionResult` implementation (for example, `object`) also returns a Formatted Response.
+
+Some helper methods of this type include `BadRequest`, `CreatedAtRoute`, and `Ok`. Examples of these methods include `return BadRequest(modelState);`, `return CreatedAtRoute("routename", values, newobject);`, and `return Ok(value);`, respectively.
```

Comment:
Also clarify that `BadRequest` and `Ok` fit here only when passed a value. `Created`, `CreatedAtRoute` and probably a couple of other methods are always in this category because their overloads all require a value.